Firmware/src/modules/tools/temperaturecontrol/max31855.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+      This file is part of Smoothie (http://smoothieware.org/). The motion control part is heavily based on Grbl (https://github.com/simen/grbl).
+      Smoothie is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+      Smoothie is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+      You should have received a copy of the GNU General Public License along with Smoothie. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <math.h>
+#include <fcntl.h>
+
+#include "lpc43_ssp.h"
+#include "max31855.h"
+#include "OutputStream.h"
+
+#define chip_select_pin_key "chip_select_pin"
+#define spi_channel_key         "spi_channel"
+#define designator_key           "designator"
+#define tool_id_key                 "tool_id"
+
+//Define maximum number of max31855 sensors which may be configured
+//TODO this defined number should be removed as we would like to configure any number of sensors
+#define NUM_MAX31855_SENSORS 2
+Max31855 *Max31855::instance[NUM_MAX31855_SENSORS];
+uint8_t Max31855::instance_index = 0;
+RingBuffer *Max31855::queue[NUM_MAX31855_SENSORS];
+
+/* The following functions outside class are provided for Nuttx:
+ * .select functions control GPIO Chip Select pin during data acquisition for target sensor
+ * .status functions perform status operations (not currently used but need to be here)
+ */
+
+// For enabled SSP0 channel
+void lpc43_ssp0select(FAR struct spi_dev_s *dev, enum spi_dev_e devid, bool selected)
```

Comment:
Sorry but h/w specific stuff does not go into main code, it goes into HAL layer. Are these standard NUTTX callbacks or hacks into nuttx?

Reply:
yes, we control chip select configured in smoothie through these functions, they are external functions from nuttx and used for max31855 module

Reply:
WHY?